src/java.desktop/share/classes/javax/swing/border/EtchedBorder.java

```diff
@@ -131,22 +150,41 @@ public EtchedBorder(int etchType, Color highlight, Color shadow)    {
      * @param height the height of the painted border
      */
     public void paintBorder(Component c, Graphics g, int x, int y, int width, int height) {
-        int w = width;
-        int h = height;
+        // We remove any initial transforms to prevent rounding errors
+        // when drawing in non-integer scales
+        AffineTransform at = new AffineTransform();
+        Stroke oldStk = new BasicStroke();
+        int stkWidth = 1;
+        if (g instanceof Graphics2D) {
+            at = ((Graphics2D) g).getTransform();
+            ((Graphics2D) g).setTransform(new AffineTransform());
+            oldStk = ((Graphics2D) g).getStroke();
+            stkWidth = (int) Math.floor(Math.min(at.getScaleX(),at.getScaleY()));
+            ((Graphics2D) g).setStroke(new BasicStroke((float) stkWidth));
+        }
```

Comment:
This reminds me .. did you try STROKE_PURE ? If so does it make any difference ?
https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/java/awt/RenderingHints.html#KEY_STROKE_CONTROL

Reply:
I don't think setting it to STROKE_PURE changed anything